### PR TITLE
Claims manager

### PR DESCRIPTION
**Pull request by Ant-Nazif**
--------------------------------------------------------------------
 * Created claims manager layout
 * Created move to category layout
 * Memoized checkIfClaimSelected
 * Changed updateCategoryForClaim to take a list of claims
 * Made small adjustments to the seed attestations
 * Display the name of the attested claim insted of attestationId
 * Added functionality to move claim/claims into different categories
--------------------------------------------------------------------

### CODE REVIEW

File: src/actions/actionCreators.js [hunk 1]
```diff
@@ -76,6 +76,12 @@ import {
   SET_SHOW_EMPTY_CLAIM_CATEGORIES,
   ADD_NEW_CATEGORY,
   SET_NEW_CATEGORY,
+  SET_CLAIM_VISIBILITY,
+  SET_SHOW_HIDDEN_CLAIMS,
+  UPDATE_SELECTED_CLAIMS,
+  CLEAR_SELECTED_CLAIMS,
+  UPDATE_CATEGORY_FOR_CLAIMS,
+  MOVE_CLAIMS_TO_CATEGORY,
```
Comment (hanzah) [inline src/actions/actionCreators.js]: Why do we keep all action creators in one file. We can split this file up based on domain, for example claims, categories, identity etc.
You can even maintain the import path just by creating an actionCreators directory and moving the exports into the index.js of it.

File: src/actions/actionCreators.js [hunk 2]
```diff
@@ -447,3 +458,27 @@ export const setNewCategory = (category) => ({
   type: SET_NEW_CATEGORY,
   payload: { category },
 });
+
+export const setClaimVisibility = (claim, value) => ({
+  type: SET_CLAIM_VISIBILITY,
+  payload: { claim, value },
+});
+
+export const updateSelectedClaims = (claim) => ({
```
Comment (hanzah) [inline src/actions/actionCreators.js]: If you name it in plural, you would expect that the action creator takes an array of claims as a parameter and not one claim.

File: src/actions/actionCreators.js [hunk 3]
```diff
@@ -447,3 +458,27 @@ export const setNewCategory = (category) => ({
...
+  type: UPDATE_SELECTED_CLAIMS,
+  payload: { claim },
+});
+
+export const clearSelectedClaims = () => ({
+  type: CLEAR_SELECTED_CLAIMS,
```
Comment (hanzah) [inline src/actions/actionCreators.js]: Maintain the footprint/signature of an action. Even if the payload is just an empty object, it should be there.

File: src/actions/actionCreators.js [hunk 4]
```diff
@@ -447,3 +458,27 @@ export const setNewCategory = (category) => ({
...
+});
+
+export const updateCategoryForClaims = (claims, categoryId) => ({
+  type: UPDATE_CATEGORY_FOR_CLAIMS,
+  payload: { claims, categoryId },
+});
+
+export const moveClaimsToCategory = (targetCategory) => ({
```
Comment (hanzah) [inline src/actions/actionCreators.js]: What claims are you moving? If it's something from the state, make is visible in the name of the action creator.

File: src/containers/Identity/PersonalInfo/ClaimCategoryDetails/View.js [hunk 5]
```diff
@@ -10,16 +10,23 @@ const ClaimCategoryDetails = (props) => {
...
     navigation.navigate('ClaimDetails', { id: claim.get('id', ''), claimName: claim.get('name', '') });
   };
 
+
   return (
     <View style={styles.root}>
+      <CheckBox
+        checked={showHiddenClaims}
+        title="Show hidden claims"
+        onPress={() => setShowHiddenClaims(!showHiddenClaims)}
```
Comment (hanzah) [inline src/containers/Identity/PersonalInfo/ClaimCategoryDetails/View.js]: If you are only using this for this kind of use case. You can name the action toggleHiddenClaims and do all the logic in the saga.

File: src/containers/Identity/PersonalInfo/ClaimManager/MoveIntoCategory/styles.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { StyleSheet } from 'react-native';
+
+
+export default StyleSheet.create({
+  root: {
+    flex: 1,
+    padding:16,
+  },
+  text:{
+    color:'#b5b5b5',
```
Comment (hanzah) [inline src/containers/Identity/PersonalInfo/ClaimManager/MoveIntoCategory/styles.js:10]: Think about extracting the colors in a top theme styles file, so we can adjust it in one place. Of course if this is a color that is dependent of the theme.

File: src/containers/Identity/PersonalInfo/ClaimManager/View.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    navigation.navigate('MoveIntoCategory');
+  };
+
+  const selectClaim = (selectedClaim) => {
+    updateSelectedClaims(selectedClaim);
+  };
+
+  const checkIfClaimIsSelected = useCallback((claim) => selectedClaims.includes(claim), [selectedClaims]);
+
+  const getCategotyName = (name) => {
```
Comment (hanzah) [inline src/containers/Identity/PersonalInfo/ClaimManager/View.js]: This is an util function, move it outside of the component.

File: src/containers/Identity/PersonalInfo/ClaimManager/View.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+      <ScrollView>
+        <View style={styles.claimsContainer}>
+          {claims.keySeq().map((claim) => (
+            <View style={styles.claims} key={claims.getIn([claim, 'id'])}>
+              <CheckBox
+                key={claims.getIn([claim, 'id'])}
+                checked={checkIfClaimIsSelected(claims.get(claim))}
+                onPress={() => selectClaim(claims.get(claim, IMap()))}
+                title={claims.getIn([claim, 'name'])}
+                containerStyle={{ backgroundColor:'white' }}
```
Comment (hanzah) [inline src/containers/Identity/PersonalInfo/ClaimManager/View.js]: Move into styles

File: src/containers/Identity/PersonalInfo/ClaimManager/View.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+              />
+              <TouchableOpacity onPress={() => moveSingleClaim(claims.get(claim))}>
+                <View style={styles.icon}>
+                  <Text style={styles.text}>{getCategotyName(claimCategories.getIn([claims.getIn([claim, 'categoryId'], ''), 'name'], ''))}</Text>
+                  <MaterialIcons name="arrow-drop-down" size={24} />
+                </View>
+              </TouchableOpacity>
+              <TouchableOpacity onPress={() => toggleClaimVisibility(claims.get(claim))}>
+                {claims.getIn([claim, 'hidden']) ? <Icon name={Platform.OS === 'ios' ? 'ios-eye-off' : 'md-eye-off'} size={24} />
+                  : <Icon name={Platform.OS === 'ios' ? 'ios-eye' : 'md-eye'} size={24} />
```
Comment (hanzah) [inline src/containers/Identity/PersonalInfo/ClaimManager/View.js]: These can be separate components, so you can reuse them whenever you need them in the app.
Size can be the prop.

<Icon name={Platform.OS === 'ios' ? 'ios-eye-off' : 'md-eye-off'} size={24} />
<Icon name={Platform.OS === 'ios' ? 'ios-eye' : 'md-eye'} size={24} />

File: src/containers/Identity/PersonalInfo/ClaimManager/styles.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+export default StyleSheet.create({
+  root: {
+    flex: 1,
+    padding:16,
+    backgroundColor:'white',
+  },
+  button: {
+    backgroundColor:
+            '#b5b5b5',
```
Comment (hanzah) [inline src/containers/Identity/PersonalInfo/ClaimManager/styles.js:12]: Same thing and code styling. The linter should catch this.

File: src/containers/Identity/PersonalInfo/View.js [hunk 11]
```diff
@@ -72,17 +72,30 @@ const PersonalInfo = (props) => {
...
+        .filter((item) => item.get('name').toUpperCase().includes(search))
+        .keySeq()
+        .map((item) => claims.get(item));
+
+      const itemDataId = claimCategory.get('id', '');
+      const itemDataName = claimCategory.get('name', '').toUpperCase();
+
+      if (itemDataName.includes(search)) {
+        return itemDataName.includes(search);
+      }
```
Comment (hanzah) [inline src/containers/Identity/PersonalInfo/View.js]: if (itemDataName.includes(search)) {
        return true;
      }

File: src/containers/Identity/PersonalInfo/View.js [hunk 12]
```diff
@@ -72,17 +72,30 @@ const PersonalInfo = (props) => {
 
   const updateSearch = (searchValue) => {
     const newData = claimCategories.filter((claimCategory) => {
```
Comment (hanzah) [inline src/containers/Identity/PersonalInfo/View.js]: Please add a comment what this function is doing and how. Think about moving it into a util function just pass into it whatever it needs.

File: src/sagas/identity.js [hunk 13]
```diff
@@ -170,14 +185,27 @@ function * handleAddNewCategory(action) {
...
       identity: selectedIdentityId,
-      claims: [],
     },
   };
 
   yield put(setNewCategory(newCategory[newCategoryId]));
   yield call(updateClaimCategoryStorage);
 }
 
+function * handleMoveClaims(action) {
```
Comment (hanzah) [inline src/sagas/identity.js:265]: Start thinking about how the API will be integrated an error handling, so your code can anticipate it easily.

File: src/selectors/identity.js [hunk 14]
```diff
@@ -21,28 +21,44 @@ export const selectClaimCategoriesObj = (state) => state.identity.getIn(['person
...
     if (!showEmptyClaimCategories) {
-      return claimCategories.filter((claimCategory) => claimCategory.get('claims', IList()).size > 0);
+      return claimCategories.filter((claimCategory) => claims.some((claim) => claim.get('categoryId', '') === claimCategory.get('id', '')));
     }
     return claimCategories;
   },
 );
 
-const selectClaims = (state) => state.identity.getIn(['personalInformation', 'claims', 'byId'], IMap());
+export const selectClaimsObj = (state) => state.identity.getIn(['personalInformation', 'claims'], IMap());
```
Comment (hanzah) [inline src/selectors/identity.js]: Think about a more descriptive name, maybe something like selectClaimsReducerState or something down that line.
Obj is just to ambiguous.

File: src/selectors/identity.js [hunk 15]
```diff
@@ -21,28 +21,44 @@ export const selectClaimCategoriesObj = (state) => state.identity.getIn(['person
...
-  [selectActiveClaimCategoryId, selectActiveCategory, selectClaims],
-  (activeCategoryId, activeCategory, claims) => claims.filter((claim) => activeCategory.getIn([activeCategoryId, 'claims'], IList()).indexOf(claim.get('id', '')) > -1),
+  [selectActiveClaimCategoryId, selectClaims],
+  (activeCategoryId, claims) => claims.filter((claim) => claim.get('categoryId', '') === activeCategoryId)
+);
+
+export const selectClaimsToDisplay = createSelector(
+  [selectClaimsByCategoryId, selectShowHiddenClaims],
+  (selectedClaims, showHiddenClaims) => {
+    if (!showHiddenClaims) {
```
Comment (hanzah) [inline src/selectors/identity.js]: Why the reverse, you are just adding an additional operation
if(showHiddenClaims) {
  return selectedClaims
}
....

File: src/selectors/identity.js [hunk 16]
```diff
@@ -21,28 +21,44 @@ export const selectClaimCategoriesObj = (state) => state.identity.getIn(['person
...
+      return selectedClaims.filter((claim) => claim.get('hidden', '') === false);
```
Comment (hanzah) [inline src/selectors/identity.js]: selectedClaims.filter((claim) => !claim.get('hidden'));

File: src/utils/asyncStore/identityStorage.js [hunk 17]
```diff
@@ -89,30 +89,39 @@ export const updateClaimCategories = async (updatedCategories) => {
...
-    await AsyncStorage.setItem('claims', JSON.stringify({ claims: seededClaims }));
+    const result = await AsyncStorage.getItem('claims');
+    return result ? JSON.parse(result).claims : [];
   } catch (error) {
     console.log(error);
   }
 };
 
-export const getIdentities = async () => {
+export const storeSeedClaims = async (identityId) => {
```
Comment (hanzah) [inline src/utils/asyncStore/identityStorage.js]: Please add a comment what this function is doing and all the cases it is handling.

File: src/utils/asyncStore/identityStorage.js [hunk 18]
```diff
@@ -143,8 +152,9 @@ export const updateIdentities = async (updatedIdentities) => {
 export const updateClaims = async (updatedClaims) => {
```
Comment (hanzah) [inline src/utils/asyncStore/identityStorage.js:126]: You should not be doing these calculations in the storage repository. Same as storeSeedClaims.
The business logic needs to go into the sagas. 
These repository functions should be simple stupid, just retreiving and storing what you pass into them.

File: src/containers/Identity/PersonalInfo/View.js (hunk 11 above)
Comment (Ant-Nazif) [reply src/containers/Identity/PersonalInfo/View.js]: done

File: src/containers/Identity/PersonalInfo/View.js (hunk 12 above)
Comment (Ant-Nazif) [reply src/containers/Identity/PersonalInfo/View.js]: done

File: src/containers/Identity/PersonalInfo/ClaimManager/View.js (hunk 7 above)
Comment (Ant-Nazif) [reply src/containers/Identity/PersonalInfo/ClaimManager/View.js]: done

File: src/containers/Identity/PersonalInfo/ClaimManager/View.js (hunk 8 above)
Comment (Ant-Nazif) [reply src/containers/Identity/PersonalInfo/ClaimManager/View.js]: done

File: src/containers/Identity/PersonalInfo/ClaimManager/View.js (hunk 9 above)
Comment (Ant-Nazif) [reply src/containers/Identity/PersonalInfo/ClaimManager/View.js]: done

**Pull request merged**